sleigh/sleigh-compile/src/matcher/mod.rs

```diff
@@ -123,10 +84,16 @@ fn compare_number_of_constrained_bits(a: &MatchCase, b: &MatchCase) -> Option<Or
     let (a_token, a_context) = pattern_mask(a);
     let (b_token, b_context) = pattern_mask(b);
     match (compare_bits_set(a_token, b_token), compare_bits_set(a_context, b_context)) {
-        (Some(Ordering::Equal), Some(Ordering::Equal)) => Some(Ordering::Equal),
-        (Some(x), Some(Ordering::Equal)) | (Some(Ordering::Equal), Some(x)) => Some(x),
-        (Some(a), Some(b)) if a == b => Some(a),
-        _ => None,
+        // context or tokens contains bits not set by the other
```

Comment:
I think the match statement before and after the this change are equivalent.

The first case in the original could be removed (it is redundant) -- but otherwise the second case in the original matches the third case in patch, and the third case in the original matches the last case in the patch.